pkg/identity/identity.go

```diff
@@ -36,6 +36,17 @@ type Identity struct {
 	// for faster lookup.
 	LabelArray labels.LabelArray `json:"-"`
 
+	// CIDRLabel is the primary identity label when the identity represents
+	// a CIDR. The Labels field will consist of all matching prefixes, e.g.
+	// 10.0.0.0/8
+	// 10.0.0.0/7
+	// 10.0.0.0/6
+	// [...]
+	// reserverd:world
+	//
+	// The CIDRLabel field will only contain 10.0.0.0/8
+	CIDRLabel labels.Labels `json:"-"`
```

Comment:
If an identity can represent either CIDR or Labels, why can't we reuse one of the existing fields?

Reply:
ah, I just re-read the comment. In that case, why isn't a CIDRList instead?

Reply:
What should be a CIDRList?

Reply:
The type seems completely unused

Reply:
```
type CIDRList struct {

        // list
        List []string `json:"list"`

        // revision
        Revision int64 `json:"revision,omitempty"`
}
```

What do we fill as the revision? What is the gain of this additional abstraction?

Reply:
@aanm The CIDRLabel field is exactly *one* label, e.g. `cidr:30.1.1.1/32` or `cidr:10.0.0.0/8`. It is no list. It represents the CIDR identity as the user would expect it.

Reply:
1) how can it be one label? Does it mean we have an identity per CIDR?
2) What about the "except" attribute that we use in the toCIDR? for example:
```
cidr:
  10.0.0.0/8
except:
  10.0.0.127/32
```
how will this translate into an identityLabels?

Reply:
also, if it is a single label why not using `labels.Label` instead of `labels.Labels`?

Reply:
> * how can it be one label? Does it mean we have an identity per CIDR?

We have an identity for each CIDR requested by the user. If a user has policy rules which whitelist `10.0.0.0/8` and `10.1.1.1/32` then we have two identities. So it makes sense to report the identity as `cidr:10.0.0.0/8` and `cidr:10.1.1.1/32`

> 2\. What about the "except" attribute that we use in the toCIDR? for example:

Need to double check this but I think this is solved via the identities themselves.

CNP:
```
apiVersion: "cilium.io/v2"
kind: CiliumNetworkPolicy
metadata:
  name: "cidr-rule"
spec:
  endpointSelector:
    matchLabels:
      name: echo
  egress:
  - toCIDR:
    - 20.1.1.1/32
  - toCIDRSet:
    - cidr: 10.0.0.0/8
      except:
      - 10.96.0.0/12
```

CEP
```
      - identity: 16777218
        identity-labels:
          cidr:20.1.1.1/32: ""
      - identity: 16777219
        identity-labels:
          cidr:10.128.0.0/9: ""
      - identity: 16777220
        identity-labels:
          cidr:10.0.0.0/10: ""
      - identity: 16777221
        identity-labels:
          cidr:10.64.0.0/11: ""
      - identity: 16777222
        identity-labels:
          cidr:10.112.0.0/12: ""
```